hooks/useInternalRouter.ts

```
@@ -7,29 +7,58 @@ export const useInternalRouter = () => {
-          searchParams.set(key, String(value));
+    ) => {
+      const { params, query } = options ?? {};
+
+      let url = basePath;
+      if (params && params.length > 0) {
+        url += '/' + params.map(String).join('/');
+      }
+
+      if (query && Object.keys(query).length > 0) {
+        const searchParams = new URLSearchParams();
+        for (const [key, value] of Object.entries(query)) {
```

ESLint

[error] Object.entries 대신 Object.keys나 Object.values 사용을 고려하세요.

(no-restricted-syntax)